python/graphstorm/model/utils.py

```diff
@@ -190,24 +196,36 @@ def save_sparse_emb(model_path, sparse_emb, ntype):
     num_embs = sparse_emb.num_embeddings
     start, end = _get_sparse_emb_range(num_embs, rank, world_size)
     # collect sparse_emb in a iterative way
-    embs = []
-    batch_size = 10240
-    # TODO: dgl.distributed.DistEmbedding should provide emb.shape
 
-    idxs = th.split(th.arange(start=start, end=end), batch_size, dim=0)
-    for idx in idxs:
-        # TODO: dgl.distributed.DistEmbedding should allow some basic tensor ops
-        embs.append(sparse_emb._tensor[idx])
-
-    embs = th.cat(embs, dim=0)
     # In distributed mode where uses an NFS folder, directly call this makedirs method to
     # create spare embedding path will cause folder permission error that prevents
     # non-rank 0 process from saving embeddings. Therefore, need rank 0 process to call
     # the create_sparse_embeds_path() method first before calling save_sparse_embeds().
     emb_path = os.path.join(model_path, ntype)
     os.makedirs(emb_path, exist_ok=True)
-    emb_file_path = os.path.join(emb_path, f'sparse_emb_{pad_file_index(rank)}.pt')
-    th.save(embs, emb_file_path)
+
+    if isinstance(sparse_emb, WholeGraphSparseEmbedding):
+        (local_tensor, _) = sparse_emb.weight.get_local_tensor(host_view=True)
+        # Using WholeGraph will save sparse emb in binary format (evenly distributed)
+        # Example: wg_sparse_emb_part_1_of_2, wg_sparse_emb_part_2_of_2
+        assert (
+            local_tensor.shape[0] == end - start
```

Comment:
Is _get_sparse_emb_range compatible with WholeGraph tensor split?

Reply:
Yes, it should. Added an assertion here for sanity check.